### ISSUE TITLE
GitBucket API Insufficient for Dependabot Script

### ISSUE THREAD

**Issue by iamjpotts** [gitbucket/gitbucket#3409]
### Before submitting an issue to GitBucket I have first:

- [X] read the [contribution guidelines](https://github.com/gitbucket/gitbucket/blob/master/.github/CONTRIBUTING.md)
- [X] searched for similar already existing issue
- [X] read the documentation and [wiki](https://github.com/gitbucket/gitbucket/wiki)

## Issue
**Impacted version**: `4.40` and earlier

**Deployment mode**: Docker container, with an Nginx reverse proxy for TLS termination (TLS required by Dependabot client)

Issue: GitBucket API support is insufficient to use Dependabot to automatically create PRs for dependency upgrades

1. Using Docker Compose or individual containers, start GitBucket, and start Nginx (or any other TLS terminating reverse proxy) to accept an https connection and proxy it to GitBucket.
1. In GitBucket, in the `Profile` of `root`, under `Applications`, create a new token
1. In GitBucket, create a new repository
1. Clone that repository, and add a project to it with a stale dependency
1. Run [dependabot-script](https://github.com/dependabot/dependabot-script) with `GITHUB_ENTERPRISE_HOSTNAME` set to the DNS name of the TLS terminating reverse proxy, and with `GITHUB_ENTERPRISE_ACCESS_TOKEN` set to the value from the previous step, and with `PROJECT_PATH` and `PACKAGE_MANAGER` set
1. Observe in the log for the `GitBucket` the entry `WARN  g.c.c.a.ApiGitReferenceControllerBase - git/refs/ endpoint may not be compatible with GitHub API v3. Consider using git/ref/ endpoint instead`
1. Observe in the log for `dependabot-script` the entry `dependabot-script/.bundle/gems/octokit-5.2.0/lib/octokit/response/raise_error.rb:14:in on_complete: POST https://dt2-22.foobarquxbaz.com/api/v3/repos/root/testing/git/trees: 404 - Not Found (Octokit::NotFound)`

The dependabot script has failed due to a missing API in GitBucket.

**Comment by takezoe** [comment]
Does that mean Create a tree API (`POST /api/v3/repos/<user>/<repo>/git/trees`) needs to be implemented to support dependabot?

**Comment by iamjpotts** [comment]
> In short, Create a tree API (`POST /api/v3/repos/<user>/<repo>/git/trees`) needs to be implemented to support dependabot?

Yes, at a minimum (I'm inferring that from the error). I don't know what other APIs would be required, if any. I suspect there would be more.

**Comment by iamjpotts** [comment]
> [@iamjpotts](https://github.com/iamjpotts) [@takezoe](https://github.com/takezoe) patched dependabot for use with bitbucket https://hub.docker.com/repository/docker/alexkekiy/dependabot-for-gitbucket/general

This link appears to be private; it at least requires a docker login.
